diff --git a/include/meson.build b/include/meson.build
index 996c9b9..16401ba 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -343,8 +343,9 @@ conf_data.set('WSCONS_SUPPORT', supports_wscons ? '1' : false)
 conf_data.set('XSERVER_LIBPCIACCESS', get_option('pciaccess') ? '1' : false)
 if get_option('pciaccess')
     pciaccess_dep = dependency('pciaccess', required: build_xorg)
-    conf_data.set('HAVE_PCI_DEVICE_IS_BOOT_DISPLAY',
-                  cc.has_function('pci_device_is_boot_display', dependencies: pciaccess_dep) ? '1' : false)
+    if cc.has_function('pci_device_is_boot_display', dependencies: pciaccess_dep)
+        conf_data.set('HAVE_PCI_DEVICE_IS_BOOT_DISPLAY', 1)
+    endif
 endif
 conf_data.set('XSERVER_PLATFORM_BUS', build_udev_kms ? '1' : false)
 conf_data.set('XSERVER_SCREEN_VRR', '1')
